code/__DEFINES/atmospherics.dm

```diff
@@ -175,4 +175,12 @@
 #define LAVALAND_EQUIPMENT_EFFECT_PRESSURE 50 //what pressure you have to be under to increase the effect of equipment meant for lavaland
 #define LAVALAND_DEFAULT_ATMOS "o2=14;n2=23;TEMP=300"
 
-#define THERMAL_ENERGY(gas) (gas.temperature * gas.heat_capacity())
\ No newline at end of file
+#define THERMAL_ENERGY(gas) (gas.temperature * gas.heat_capacity())
+
+//ASSERT_GAS(gas_mixture,gas_id) - used to guarantee that the gas list for this id exists in gas_mixture.
+//Must be used before adding to a gas. May be used before reading from a gas.
+#define ASSERT_GAS(gas_mixture,gas_id) if (!gas_mixture.gases[gas_id]) { GASLIST(gas_id, gas_mixture.gases[gas_id]) };

```

Comment:
wait, how does this compile if it's above GASLIST?
byond, why are you strange.

Reply:
wait wtf